std/assembly/array.ts

```diff
@@ -496,7 +501,7 @@ export class Array<T> {
 
   flat(): T {
     if (!isArray<T>()) {
-      ERROR("Cannot call flat() on Array<T> where T is not an Array.");
+      throw new TypeError("Cannot call flat() on Array<T> where T is not an Array.");
```

Comment:
Let's move the error message to `util/error.ts` along the other ones so we can reuse it. Suggesting something short like:

```ts
export const E_ILLEGALTYPE: string = "Illegal type";
```

Reply:
Done!